crates/iota-framework/packages/iota-system/sources/validator_set.move

```diff
@@ -1129,4 +1490,153 @@ module iota_system::validator_set {
         };
         res
     }
+
+    public(package) fun committee_validator_addresses(self: &ValidatorSetV2): vector<address> {
+        let vs = &self.active_validators;
+        let num_validators = vs.length();
+        let committee_members = &self.committee_members;
+
+        let mut res = vector[];
+        let mut i = 0;
+        let committee_members_num = committee_members.length();
+        while (i < committee_members_num) {
+            let validator_index = committee_members[i];
+            assert!(
+                validator_index < num_validators,
+                ECommitteeMembersSetCorrupt,
+            );
+            let validator_address = vs[validator_index].iota_address();
+            res.push_back(validator_address);
+            i = i + 1;
+        };
+        res
+    }
+
+    // Selects new set committee members from the set of all active validators.
+    // Currently selects all active validators to be part of the committee.
+    public(package) fun select_committee_members(self: &mut ValidatorSetV2) {
+        let mut new_committee_members = vector[];
+        let active_validators = &self.active_validators;
+        let active_validators_num = active_validators.length();
+        
+        let mut i = 0;
+        while (i < active_validators_num) {
+            new_committee_members.push_back(i);
+            i = i + 1;
+        };
+
+        self.committee_members = new_committee_members;
+    }
+
+
+    // Selects top N stakers among all active validators to be part of the committee.
+    public(package) fun top_n_stakers(self: &mut ValidatorSetV2, n: u64) {
+        let validators_num = self.active_validators.length();
+
+        // Create a vector of indices
+        let mut validator_indices = vector[];
+        let mut validator_stakes = vector[];
+
+        // Fill the indices and stakes vector.
+        let mut i = 0;
+        while (i < validators_num) {
+            validator_indices.push_back(i);
+            validator_stakes.push_back(self.active_validators[i].total_stake()); // TODO: based on what else do we want to sort?
+
+            i = i + 1;
+        };
+
+        // Sort indices based on the stake values 
+        let mut i = 1;
+        while (i < validators_num) {
```

Comment:
Is insertion sort fine complexity wise in this case: O(n²)?
Or can we somehow implement Quicksort: O(n log n) on average case?

Reply:
That's an improvement we can work on. The number of validators will not be huge, so it will not have a big impact.

Reply:
Why not use priority_queue?

Reply:
Priority queue can only work with a single value and I've just implemented tie-breaking by comparing authority_key in case of equal stake. Unfortunately tuples are not comparable like in Rust.